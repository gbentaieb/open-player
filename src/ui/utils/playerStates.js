import * as states from '../constants/CoreStates';

export const playingStates = [states.PLAYING, states.BUFFERING, states.SEEKING];
export const waitingStates = [states.LOADING, states.BUFFERING, states.SEEKING];

export function isWaiting(state) {
  return waitingStates.includes(state);
}

export function isPlaying(state) {
  return playingStates.includes(state);
}
