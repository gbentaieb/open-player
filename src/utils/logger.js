/* eslint-disable no-console */

class Logger {
  defaultLevel = 5;

  LEVELS = {
    LOG: 0,
    DEBUG: 1,
    INFO: 2,
    WARN: 3,
    ERROR: 4,
    OFF: 5,
  };

  constructor() {
    this.currentLevel = this.defaultLevel;
  }

  static get instance() {
    if (!this.singleton) {
      this.singleton = new Logger();
    }

    return this.singleton;
  }

  setLevel(value) {
    if (typeof value === 'string') {
      this.currentLevel = this.LEVELS[value];
    } else {
      this.currentLevel = value;
    }
  }

  isEnabled(level) {
    return level >= this.currentLevel;
  }

  log(msg, ...args) {
    const { LOG } = this.LEVELS;

    if (this.isEnabled(LOG)) {
      console.log(msg, ...args);
    }
  }

  debug(msg, ...args) {
    const { DEBUG } = this.LEVELS;

    if (this.isEnabled(DEBUG)) {
      console.debug(msg, ...args);
    }
  }

  info(msg, ...args) {
    const { INFO } = this.LEVELS;

    if (this.isEnabled(INFO)) {
      console.info(msg, ...args);
    }
  }

  warn(msg, ...args) {
    const { WARN } = this.LEVELS;

    if (this.isEnabled(WARN)) {
      console.warn(msg, ...args);
    }
  }

  error(msg, ...args) {
    const { ERROR } = this.LEVELS;

    if (this.isEnabled(ERROR)) {
      console.error(msg, ...args);
    }
  }
}

const logger = Logger.instance;

export default logger;
