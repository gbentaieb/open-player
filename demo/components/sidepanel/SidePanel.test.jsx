import React from 'react';
import { stub } from 'sinon';
import SidePanel from './SidePanel';
import { MuiMountWithContext } from '../../../test/utils';

test('SidePanel > Snapshot', () => {
  const onClickOutside = () => {};
  const wrapper = shallow(<SidePanel onClickOutside={onClickOutside} />);
  expect(wrapper).toMatchSnapshot();
});

test('SidePanel > handleClickOutside', () => {
  const onClickOutside = stub();
  const wrapper = MuiMountWithContext(<SidePanel onClickOutside={onClickOutside} />);
  wrapper.instance().handleClickOutside({ target: document.createElement('div') });
  expect(onClickOutside.calledOnce).toEqual(true);
});
