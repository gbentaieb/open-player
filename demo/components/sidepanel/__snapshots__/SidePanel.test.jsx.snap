// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test demo sidepanel SidePanel > Snapshot 1`] = `
<div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        color="inherit"
        variant="h6"
      >
        Open Player Demo
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(List)
    component="nav"
  >
    <WithStyles(ListItem)
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ListItemText)
        primary="Play"
      />
    </WithStyles(ListItem)>
    <WithStyles(ListItem)
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ListItemText)
        primary="Pause"
      />
    </WithStyles(ListItem)>
    <WithStyles(Divider) />
    <WithStyles(ListItem)
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <GithubIcon />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        primary="Github repository"
      />
    </WithStyles(ListItem)>
  </WithStyles(List)>
</div>
`;
